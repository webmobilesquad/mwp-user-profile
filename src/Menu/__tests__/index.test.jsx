import render from 'preact-render-spy';
import {
  LogoutIcon,
  Menu,
  MenuHeader,
  MenuItem,
  MenuItems,
  MenuItemText,
  ProfileIcon,
} from '../index';

describe('<Menu />', () => {
  test('<Menu /> matches snapshot', () => {
    expect(render(<Menu />)).toMatchSnapshot();
  });
  test('<Menu visible /> matches snapshot', () => {
    expect(render(<Menu visible />)).toMatchSnapshot();
  });
  test('<Menu anchor="top-left" /> matches snapshot', () => {
    expect(render(<Menu anchor="top-left" />)).toMatchSnapshot();
  });
  test('<Menu anchor="top-right" /> matches snapshot', () => {
    expect(render(<Menu anchor="top-right" />)).toMatchSnapshot();
  });
});

describe('<MenuHeader />', () => {
  test('<MenuHeader /> matches snapshot', () => {
    expect(render(<MenuHeader />)).toMatchSnapshot();
  });
});

describe('<MenuItems />', () => {
  test('<MenuItems /> matches snapshot', () => {
    expect(render(<MenuItems />)).toMatchSnapshot();
  });

  test('<MenuItems/> with <MenuItem/> matches snapshot', () => {
    expect(
      render(
        <MenuItems>
          <MenuItem href="https://mywebportal.michelin-solutions.com/profile" />
          <MenuItem href="https://mywebportal.michelin-solutions.com/logout" />
        </MenuItems>,
      ),
    ).toMatchSnapshot();
  });
});

describe('<MenuItem />', () => {
  test('<MenuItem /> matches snapshot', () => {
    expect(
      render(<MenuItem href="https://mywebportal.michelin-solutions.com/msn/profile" />),
    ).toMatchSnapshot();
  });

  test('<MenuItem /> with text and icon matches snapshot', () => {
    expect(
      render(
        <MenuItem href="https://mywebportal.michelin-solutions.com/msn/profile">
          <ProfileIcon />
          <MenuItemText>Profile</MenuItemText>
        </MenuItem>,
      ),
    ).toMatchSnapshot();
  });
});

describe('<MenuItemText />', () => {
  test('<MenuItemText /> matches snapshot', () => {
    expect(render(<MenuItemText>Logout</MenuItemText>)).toMatchSnapshot();
  });
});

describe('<ProfileIcon />', () => {
  test('<ProfileIcon /> matches snapshot', () => {
    expect(render(<ProfileIcon />)).toMatchSnapshot();
  });
});

describe('<LogoutIcon />', () => {
  test('<LogoutIcon /> matches snapshot', () => {
    expect(render(<LogoutIcon />)).toMatchSnapshot();
  });
});
