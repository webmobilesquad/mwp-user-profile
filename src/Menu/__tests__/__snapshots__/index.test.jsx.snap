// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LogoutIcon /> <LogoutIcon /> matches snapshot 1`] = `
preact-render-spy (1 nodes)
-------
<svg
  xmlns="http://www.w3.org/2000/svg"
  width="24"
  height="24"
  viewBox="0 0 24 24"
  aria-hidden="true"
>
  <path
    d="M0 0h24v24H0z"
    fill="none"
  >
  </path>
  <path d="M10.09 15.59L11.5 17l5-5-5-5-1.41 1.41L12.67 11H3v2h9.67l-2.58 2.59zM19 3H5c-1.11 0-2 .9-2 2v4h2V5h14v14H5v-4H3v4c0 1.1.89 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"></path>
</svg>

`;

exports[`<Menu /> <Menu /> matches snapshot 1`] = `
preact-render-spy (1 nodes)
-------
<div class="Menu Menu__anchor_top-right"></div>

`;

exports[`<Menu /> <Menu anchor="top-left" /> matches snapshot 1`] = `
preact-render-spy (1 nodes)
-------
<div class="Menu Menu__anchor_top-left"></div>

`;

exports[`<Menu /> <Menu anchor="top-right" /> matches snapshot 1`] = `
preact-render-spy (1 nodes)
-------
<div class="Menu Menu__anchor_top-right"></div>

`;

exports[`<Menu /> <Menu visible /> matches snapshot 1`] = `
preact-render-spy (1 nodes)
-------
<div class="Menu Menu__visible Menu__anchor_top-right"></div>

`;

exports[`<MenuHeader /> <MenuHeader /> matches snapshot 1`] = `
preact-render-spy (1 nodes)
-------
<header class="Menu_Header"></header>

`;

exports[`<MenuItem /> <MenuItem /> matches snapshot 1`] = `
preact-render-spy (1 nodes)
-------
<li class="Menu_Item">
  <a href="https://mywebportal.michelin-solutions.com/msn/profile"></a>
</li>

`;

exports[`<MenuItem /> <MenuItem /> with text and icon matches snapshot 1`] = `
preact-render-spy (1 nodes)
-------
<li class="Menu_Item">
  <a href="https://mywebportal.michelin-solutions.com/msn/profile">
    <svg
      xmlns="http://www.w3.org/2000/svg"
      width="24"
      height="24"
      viewBox="0 0 24 24"
      aria-hidden="true"
    >
      <path d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 3c1.66 0 3 1.34 3 3s-1.34 3-3 3-3-1.34-3-3 1.34-3 3-3zm0 14.2c-2.5 0-4.71-1.28-6-3.22.03-1.99 4-3.08 6-3.08 1.99 0 5.97 1.09 6 3.08-1.29 1.94-3.5 3.22-6 3.22z"></path>
      <path
        d="M0 0h24v24H0z"
        fill="none"
      >
      </path>
    </svg>
    <span>Profile</span>
  </a>
</li>

`;

exports[`<MenuItemText /> <MenuItemText /> matches snapshot 1`] = `
preact-render-spy (1 nodes)
-------
<span>Logout</span>

`;

exports[`<MenuItems /> <MenuItems /> matches snapshot 1`] = `
preact-render-spy (1 nodes)
-------
<ul class="Menu_Items"></ul>

`;

exports[`<MenuItems /> <MenuItems/> with <MenuItem/> matches snapshot 1`] = `
preact-render-spy (1 nodes)
-------
<ul class="Menu_Items">
  <li class="Menu_Item">
    <a href="https://mywebportal.michelin-solutions.com/profile"></a>
  </li>
  <li class="Menu_Item">
    <a href="https://mywebportal.michelin-solutions.com/logout"></a>
  </li>
</ul>

`;

exports[`<ProfileIcon /> <ProfileIcon /> matches snapshot 1`] = `
preact-render-spy (1 nodes)
-------
<svg
  xmlns="http://www.w3.org/2000/svg"
  width="24"
  height="24"
  viewBox="0 0 24 24"
  aria-hidden="true"
>
  <path d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 3c1.66 0 3 1.34 3 3s-1.34 3-3 3-3-1.34-3-3 1.34-3 3-3zm0 14.2c-2.5 0-4.71-1.28-6-3.22.03-1.99 4-3.08 6-3.08 1.99 0 5.97 1.09 6 3.08-1.29 1.94-3.5 3.22-6 3.22z"></path>
  <path
    d="M0 0h24v24H0z"
    fill="none"
  >
  </path>
</svg>

`;
